docs/sources/developers/plugins/annotations.md

```diff
@@ -0,0 +1,137 @@
++++
+title = "Add annotations to your data source plugin"
+type = "docs"
++++
+
+# Add annotations to your data source plugin
+
+This guide explains how to add support [annotations]({{< relref "reference/annotations.md" >}}) to your data source plugin.
+
+This guide assumes that you're already familiar with how to [Build a data source plugin]({{< relref "../../../../../tutorials/build-a-data-source-plugin.md" >}}).
+
+1. Let Grafana know that your data source plugin can return annotations events, by adding `"annotations": true` to the [plugin.json]({{< relref "metadata.md" >}}) file.
+
+   ```json
+   {
+     "annotations": true
+   }
+   ```
+
+1. In `DataSource.ts`, override the `annotationQuery` method from `DataSourceApi`.
+
+   ```ts
+   async annotationQuery(options: AnnotationQueryRequest<MyQuery>): Promise<AnnotationEvent[]> {
+       return [];
+   }
+   ```
+
+1. Return an array of [AnnotationEvent]({{< relref "../../packages_api/data/annotationevent.md" >}}).
+
+   ```ts
+   async annotationQuery(options: AnnotationQueryRequest<MyQuery>): Promise<AnnotationEvent[]> {
+       const events: AnnotationEvent[] = [];
+
+       const date = new Date();
+
+       const event: AnnotationEvent = {
+         time: date.valueOf(),
+         text: 'foo',
+         tags: ['bar'],
+       };
+
+       events.push(event);
+
+       return events;
+   }
+   ```
+
+## Region annotations
+
+[Region annotations]({{< relref "../../reference/annotations.md#adding-regions-events" >}}) have a start and end time. This can for example be used to annotate maintenance windows or downtime.
+
+To return a region annotation, set the `timeEnd`, and `isRegion` properties.
+
+```ts
+const regionEvent: AnnotationEvent = {
+  time: startDate.valueOf(),
+  timeEnd: endDate.valueOf(),
+  isRegion: true,
+  text: 'foo',
+  tags: ['bar'],
+};
+```
+
+## Query editor
+
+Let users write custom annotation queries to only display the annotation events they care about, by adding a _query editor_.
+
+> **Note**: Annotation query editors have yet to receive support for React. The instructions here are given for Angular. Fortunately, you can run Angular even in a plugin otherwise written using React. This section will be updated once React support for annotation queries editors is available.
+
+1. Create a file called `AnnotationQueryEditor.ts` in the plugin root directory, with the following content:
+
+   ```ts
+   export class AnnotationQueryEditor {
+     static templateUrl = 'partials/annotations.editor.html';
+
+     annotation: any;
+
+     /** @ngInject */
+     constructor() {
+       this.annotation.rawQuery = this.annotation.rawQuery || '';
+     }
+   }
+   ```
+
+1. Create a directory called `partials` in the plugin root directory.
+
+   ```
+   mkdir partials
+   ```
+
+1. Create a file called `annotations.editor.html` in the `partials` directory you just created, with the following content:
```

Comment:
```suggestion
1. Create a file called `annotations.editor.html` in the `partials` directory you just created. Add the following content to the file:
```

Reply:
See above comment on step consistency, This is a bit different since it is calling out content to add into a file. You might want to split this into two steps.

1. Make the file called...
2. Add content to the file.

Reply:
Ignore this comment if it is actually one step in the command line editors.